controllers/services/auth/auth_controller_actions.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"errors"
+	"slices"
+	"strings"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/services/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Templates = []odhtypes.TemplateInfo{
+		{
+			FS:   resourcesFS,
+			Path: AdminGroupRoleTemplate,
+		},
+		{
+			FS:   resourcesFS,
+			Path: AllowedGroupRoleTemplate,
+		},
+		{
+			FS:   resourcesFS,
+			Path: AdminGroupClusterRoleTemplate,
+		},
+	}
+
+	return nil
+}
+
+// We only really expect this to copy once, the fields in the dashboardConfig will be immutable
+// but there may be edge cases where the dashboardConfig is created or edited later.
+// This function can be removed entirely when the dashboard team deprecates
+// the fields in question.
+func copyGroups(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	ai, ok := rr.Instance.(*serviceApi.Auth)
+	if !ok {
+		return errors.New("instance is not of type *services.Auth")
+	}
+
+	newGroups := false
+
+	// Get groups from the dashboardConfig
+	odhObject := &unstructured.Unstructured{}
+	odhObject.SetGroupVersionKind(gvk.OdhDashboardConfig)
+
+	err := rr.Client.Get(ctx, client.ObjectKey{
+		Name:      "odh-dashboard-config",
+		Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+	}, odhObject)
+	if err != nil && !k8serr.IsNotFound(err) {
+		return err
+	}
+	foundGroups, found, _ := unstructured.NestedStringMap(odhObject.Object, "spec", "groupsConfig")
+	if !found {
+		return errors.New("no groupsConfig found in dashboardConfig")
```

Comment:
is it fatal?

Reply:
No, if the content isn't there we can continue, we just don't copy in any content.